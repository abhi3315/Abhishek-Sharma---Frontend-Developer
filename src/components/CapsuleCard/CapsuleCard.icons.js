import propTypes from 'prop-types';

function Status({ color }) {
	return (
		<svg
			xmlns='http://www.w3.org/2000/svg'
			fill='none'
			viewBox='0 0 24 24'
			strokeWidth='1.5'
			stroke='currentColor'
			className={`w-6 h-6 ${color}`}
		>
			<path
				strokeLinecap='round'
				strokeLinejoin='round'
				d='M3 13.125C3 12.504 3.504 12 4.125 12h2.25c.621 0 1.125.504 1.125 1.125v6.75C7.5 20.496 6.996 21 6.375 21h-2.25A1.125 1.125 0 013 19.875v-6.75zM9.75 8.625c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125v11.25c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V8.625zM16.5 4.125c0-.621.504-1.125 1.125-1.125h2.25C20.496 3 21 3.504 21 4.125v15.75c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V4.125z'
			/>
		</svg>
	);
}

Status.propTypes = {
	color: propTypes.string.isRequired,
};
function Reuse() {
	return (
		<svg
			xmlns='http://www.w3.org/2000/svg'
			fill='none'
			viewBox='0 0 24 24'
			strokeWidth={1.5}
			stroke='currentColor'
			className='w-6 h-6'
		>
			<path
				strokeLinecap='round'
				strokeLinejoin='round'
				d='M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0l3.181 3.183a8.25 8.25 0 0013.803-3.7M4.031 9.865a8.25 8.25 0 0113.803-3.7l3.181 3.182m0-4.991v4.99'
			/>
		</svg>
	);
}

function Truck() {
	return (
		<svg
			xmlns='http://www.w3.org/2000/svg'
			fill='none'
			viewBox='0 0 24 24'
			strokeWidth={1.5}
			stroke='currentColor'
			className='w-6 h-6'
		>
			<path
				strokeLinecap='round'
				strokeLinejoin='round'
				d='M8.25 18.75a1.5 1.5 0 01-3 0m3 0a1.5 1.5 0 00-3 0m3 0h6m-9 0H3.375a1.125 1.125 0 01-1.125-1.125V14.25m17.25 4.5a1.5 1.5 0 01-3 0m3 0a1.5 1.5 0 00-3 0m3 0h1.125c.621 0 1.129-.504 1.09-1.124a17.902 17.902 0 00-3.213-9.193 2.056 2.056 0 00-1.58-.86H14.25M16.5 18.75h-2.25m0-11.177v-.958c0-.568-.422-1.048-.987-1.106a48.554 48.554 0 00-10.026 0 1.106 1.106 0 00-.987 1.106v7.635m12-6.677v6.677m0 4.5v-4.5m0 0h-12'
			/>
		</svg>
	);
}

function Lifebuoy() {
	return (
		<svg
			xmlns='http://www.w3.org/2000/svg'
			fill='none'
			viewBox='0 0 24 24'
			strokeWidth={1.5}
			stroke='currentColor'
			className='w-6 h-6'
		>
			<path
				strokeLinecap='round'
				strokeLinejoin='round'
				d='M16.712 4.33a9.027 9.027 0 011.652 1.306c.51.51.944 1.064 1.306 1.652M16.712 4.33l-3.448 4.138m3.448-4.138a9.014 9.014 0 00-9.424 0M19.67 7.288l-4.138 3.448m4.138-3.448a9.014 9.014 0 010 9.424m-4.138-5.976a3.736 3.736 0 00-.88-1.388 3.737 3.737 0 00-1.388-.88m2.268 2.268a3.765 3.765 0 010 2.528m-2.268-4.796a3.765 3.765 0 00-2.528 0m4.796 4.796c-.181.506-.475.982-.88 1.388a3.736 3.736 0 01-1.388.88m2.268-2.268l4.138 3.448m0 0a9.027 9.027 0 01-1.306 1.652c-.51.51-1.064.944-1.652 1.306m0 0l-3.448-4.138m3.448 4.138a9.014 9.014 0 01-9.424 0m5.976-4.138a3.765 3.765 0 01-2.528 0m0 0a3.736 3.736 0 01-1.388-.88 3.737 3.737 0 01-.88-1.388m2.268 2.268L7.288 19.67m0 0a9.024 9.024 0 01-1.652-1.306 9.027 9.027 0 01-1.306-1.652m0 0l4.138-3.448M4.33 16.712a9.014 9.014 0 010-9.424m4.138 5.976a3.765 3.765 0 010-2.528m0 0c.181-.506.475-.982.88-1.388a3.736 3.736 0 011.388-.88m-2.268 2.268L4.33 7.288m6.406 1.18L7.288 4.33m0 0a9.024 9.024 0 00-1.652 1.306A9.025 9.025 0 004.33 7.288'
			/>
		</svg>
	);
}

export default { Status, Reuse, Truck, Lifebuoy };
