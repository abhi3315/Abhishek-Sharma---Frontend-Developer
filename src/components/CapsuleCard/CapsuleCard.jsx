import propTypes from 'prop-types';
import { useState } from 'react';

import capsuleUtil from '../../utils/capsules';
import CapsuleModal from '../CapsuleModal';
import Icons from './CapsuleCard.icons';

function CapsuleCard({ capsule }) {
	const [openModal, setOpenModal] = useState(false);

	const handleOpenModal = () => {
		document.body.style.overflow = 'hidden';
		setOpenModal(true);
	};

	const handleCloseModal = () => {
		document.body.style.overflow = 'auto';
		setOpenModal(false);
	};

	const {
		serial,
		status,
		type,
		id,
		water_landings: waterLandings,
		land_landings: landLandings,
		reuse_count: reuseCount,
		last_update: lastUpdate,
	} = capsule;

	const statusLabels = capsuleUtil.getCapsuleStatusLabels();
	const statusColor = capsuleUtil.getStatusColor(status);

	return (
		<>
			<div
				className='flex justify-center cursor-pointer'
				aria-hidden='true'
				onClick={handleOpenModal}
			>
				<div className='block max-w-sm rounded-2xl overflow-hidden bg-white text-center shadow-lg hover:shadow-2xl transition duration-500 ease-in-out dark:bg-neutral-700 w-full'>
					<div className='p-6 h-32 overflow-hidden'>
						<h5 className='mb-1 text-xl font-medium leading-tight text-neutral-800 dark:text-neutral-50 line-clamp-1'>
							<span title='Capsule Serial'>{serial}</span> |{' '}
							<span title='Capsule Type'>{type}</span>
						</h5>
						{lastUpdate && (
							<p className='text-base text-neutral-600 dark:text-neutral-200 line-clamp-2'>
								<strong>Last Update: </strong>
								{lastUpdate}
							</p>
						)}
					</div>
					<div className='border-t-2 border-neutral-100 py-3 px-6 dark:border-neutral-600 dark:text-neutral-50 grid grid-cols-2 gap-4 pb-8'>
						<div className='flex justify-center items-center flex-col' title='Status'>
							<Icons.Status color={statusColor} />
							<p className={`line-clamp-1 ${statusColor}`}>{statusLabels[status]}</p>
						</div>
						<div className='flex justify-center items-center flex-col' title='Resuse Count'>
							<Icons.Reuse />
							<p>{reuseCount}</p>
						</div>
						<div className='flex justify-center items-center flex-col' title='Land Landings'>
							<Icons.Truck />
							<p>{landLandings}</p>
						</div>
						<div className='flex justify-center items-center flex-col' title='Water Landings'>
							<Icons.Lifebuoy />
							<p>{waterLandings}</p>
						</div>
					</div>
				</div>
			</div>
			{openModal && <CapsuleModal id={id} isOpen={openModal} onClose={handleCloseModal} />}
		</>
	);
}

CapsuleCard.propTypes = {
	capsule: propTypes.shape({
		id: propTypes.string,
		serial: propTypes.string,
		status: propTypes.string,
		type: propTypes.string,
		last_update: propTypes.string,
		water_landings: propTypes.number,
		land_landings: propTypes.number,
		reuse_count: propTypes.number,
	}),
};

CapsuleCard.defaultProps = {
	capsule: {},
};

export default CapsuleCard;
