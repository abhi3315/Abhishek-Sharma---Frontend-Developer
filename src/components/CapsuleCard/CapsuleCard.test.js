import { cleanup, render } from '@testing-library/react';

import CapsuleCard from './CapsuleCard';

afterEach(cleanup);

const capsule = {
	reuse_count: 2,
	water_landings: 3,
	land_landings: 0,
	last_update:
		'As of August 29, 2019: Arrived at Port of LA after splashdown following CRS-18 mission',
	launches: ['5eb87cecffd86e000604b33f', '5eb87d0effd86e000604b35c', '5eb87d36ffd86e000604b37b'],
	serial: 'C108',
	status: 'active',
	type: 'Dragon 1.1',
	id: '5e9e2c5cf359188bfb3b266b',
};

describe('CapsuleCard', () => {
	it('should render correctly', () => {
		const { container } = render(<CapsuleCard />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with data', () => {
		const { container } = render(<CapsuleCard capsule={capsule} />);
		expect(container).toMatchSnapshot();
	});
});
