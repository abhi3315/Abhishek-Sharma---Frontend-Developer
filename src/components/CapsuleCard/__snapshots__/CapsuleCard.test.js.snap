// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CapsuleCard should render correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    class="flex justify-center cursor-pointer"
  >
    <div
      class="block max-w-sm rounded-2xl overflow-hidden bg-white text-center shadow-lg hover:shadow-2xl transition duration-500 ease-in-out dark:bg-neutral-700 w-full"
    >
      <div
        class="p-6 h-32 overflow-hidden"
      >
        <h5
          class="mb-1 text-xl font-medium leading-tight text-neutral-800 dark:text-neutral-50 line-clamp-1"
        >
          <span
            title="Capsule Serial"
          />
           |
           
          <span
            title="Capsule Type"
          />
        </h5>
      </div>
      <div
        class="border-t-2 border-neutral-100 py-3 px-6 dark:border-neutral-600 dark:text-neutral-50 grid grid-cols-2 gap-4 pb-8"
      >
        <div
          class="flex justify-center items-center flex-col"
          title="Status"
        >
          <svg
            class="w-6 h-6 text-green-500"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 13.125C3 12.504 3.504 12 4.125 12h2.25c.621 0 1.125.504 1.125 1.125v6.75C7.5 20.496 6.996 21 6.375 21h-2.25A1.125 1.125 0 013 19.875v-6.75zM9.75 8.625c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125v11.25c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V8.625zM16.5 4.125c0-.621.504-1.125 1.125-1.125h2.25C20.496 3 21 3.504 21 4.125v15.75c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V4.125z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="line-clamp-1 text-green-500"
          />
        </div>
        <div
          class="flex justify-center items-center flex-col"
          title="Resuse Count"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0l3.181 3.183a8.25 8.25 0 0013.803-3.7M4.031 9.865a8.25 8.25 0 0113.803-3.7l3.181 3.182m0-4.991v4.99"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p />
        </div>
        <div
          class="flex justify-center items-center flex-col"
          title="Land Landings"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.25 18.75a1.5 1.5 0 01-3 0m3 0a1.5 1.5 0 00-3 0m3 0h6m-9 0H3.375a1.125 1.125 0 01-1.125-1.125V14.25m17.25 4.5a1.5 1.5 0 01-3 0m3 0a1.5 1.5 0 00-3 0m3 0h1.125c.621 0 1.129-.504 1.09-1.124a17.902 17.902 0 00-3.213-9.193 2.056 2.056 0 00-1.58-.86H14.25M16.5 18.75h-2.25m0-11.177v-.958c0-.568-.422-1.048-.987-1.106a48.554 48.554 0 00-10.026 0 1.106 1.106 0 00-.987 1.106v7.635m12-6.677v6.677m0 4.5v-4.5m0 0h-12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p />
        </div>
        <div
          class="flex justify-center items-center flex-col"
          title="Water Landings"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.712 4.33a9.027 9.027 0 011.652 1.306c.51.51.944 1.064 1.306 1.652M16.712 4.33l-3.448 4.138m3.448-4.138a9.014 9.014 0 00-9.424 0M19.67 7.288l-4.138 3.448m4.138-3.448a9.014 9.014 0 010 9.424m-4.138-5.976a3.736 3.736 0 00-.88-1.388 3.737 3.737 0 00-1.388-.88m2.268 2.268a3.765 3.765 0 010 2.528m-2.268-4.796a3.765 3.765 0 00-2.528 0m4.796 4.796c-.181.506-.475.982-.88 1.388a3.736 3.736 0 01-1.388.88m2.268-2.268l4.138 3.448m0 0a9.027 9.027 0 01-1.306 1.652c-.51.51-1.064.944-1.652 1.306m0 0l-3.448-4.138m3.448 4.138a9.014 9.014 0 01-9.424 0m5.976-4.138a3.765 3.765 0 01-2.528 0m0 0a3.736 3.736 0 01-1.388-.88 3.737 3.737 0 01-.88-1.388m2.268 2.268L7.288 19.67m0 0a9.024 9.024 0 01-1.652-1.306 9.027 9.027 0 01-1.306-1.652m0 0l4.138-3.448M4.33 16.712a9.014 9.014 0 010-9.424m4.138 5.976a3.765 3.765 0 010-2.528m0 0c.181-.506.475-.982.88-1.388a3.736 3.736 0 011.388-.88m-2.268 2.268L4.33 7.288m6.406 1.18L7.288 4.33m0 0a9.024 9.024 0 00-1.652 1.306A9.025 9.025 0 004.33 7.288"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CapsuleCard should render correctly with data 1`] = `
<div>
  <div
    aria-hidden="true"
    class="flex justify-center cursor-pointer"
  >
    <div
      class="block max-w-sm rounded-2xl overflow-hidden bg-white text-center shadow-lg hover:shadow-2xl transition duration-500 ease-in-out dark:bg-neutral-700 w-full"
    >
      <div
        class="p-6 h-32 overflow-hidden"
      >
        <h5
          class="mb-1 text-xl font-medium leading-tight text-neutral-800 dark:text-neutral-50 line-clamp-1"
        >
          <span
            title="Capsule Serial"
          />
           |
           
          <span
            title="Capsule Type"
          >
            rocket
          </span>
        </h5>
      </div>
      <div
        class="border-t-2 border-neutral-100 py-3 px-6 dark:border-neutral-600 dark:text-neutral-50 grid grid-cols-2 gap-4 pb-8"
      >
        <div
          class="flex justify-center items-center flex-col"
          title="Status"
        >
          <svg
            class="w-6 h-6 text-green-500"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 13.125C3 12.504 3.504 12 4.125 12h2.25c.621 0 1.125.504 1.125 1.125v6.75C7.5 20.496 6.996 21 6.375 21h-2.25A1.125 1.125 0 013 19.875v-6.75zM9.75 8.625c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125v11.25c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V8.625zM16.5 4.125c0-.621.504-1.125 1.125-1.125h2.25C20.496 3 21 3.504 21 4.125v15.75c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V4.125z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="line-clamp-1 text-green-500"
          />
        </div>
        <div
          class="flex justify-center items-center flex-col"
          title="Resuse Count"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0l3.181 3.183a8.25 8.25 0 0013.803-3.7M4.031 9.865a8.25 8.25 0 0113.803-3.7l3.181 3.182m0-4.991v4.99"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p />
        </div>
        <div
          class="flex justify-center items-center flex-col"
          title="Land Landings"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.25 18.75a1.5 1.5 0 01-3 0m3 0a1.5 1.5 0 00-3 0m3 0h6m-9 0H3.375a1.125 1.125 0 01-1.125-1.125V14.25m17.25 4.5a1.5 1.5 0 01-3 0m3 0a1.5 1.5 0 00-3 0m3 0h1.125c.621 0 1.129-.504 1.09-1.124a17.902 17.902 0 00-3.213-9.193 2.056 2.056 0 00-1.58-.86H14.25M16.5 18.75h-2.25m0-11.177v-.958c0-.568-.422-1.048-.987-1.106a48.554 48.554 0 00-10.026 0 1.106 1.106 0 00-.987 1.106v7.635m12-6.677v6.677m0 4.5v-4.5m0 0h-12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p />
        </div>
        <div
          class="flex justify-center items-center flex-col"
          title="Water Landings"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.712 4.33a9.027 9.027 0 011.652 1.306c.51.51.944 1.064 1.306 1.652M16.712 4.33l-3.448 4.138m3.448-4.138a9.014 9.014 0 00-9.424 0M19.67 7.288l-4.138 3.448m4.138-3.448a9.014 9.014 0 010 9.424m-4.138-5.976a3.736 3.736 0 00-.88-1.388 3.737 3.737 0 00-1.388-.88m2.268 2.268a3.765 3.765 0 010 2.528m-2.268-4.796a3.765 3.765 0 00-2.528 0m4.796 4.796c-.181.506-.475.982-.88 1.388a3.736 3.736 0 01-1.388.88m2.268-2.268l4.138 3.448m0 0a9.027 9.027 0 01-1.306 1.652c-.51.51-1.064.944-1.652 1.306m0 0l-3.448-4.138m3.448 4.138a9.014 9.014 0 01-9.424 0m5.976-4.138a3.765 3.765 0 01-2.528 0m0 0a3.736 3.736 0 01-1.388-.88 3.737 3.737 0 01-.88-1.388m2.268 2.268L7.288 19.67m0 0a9.024 9.024 0 01-1.652-1.306 9.027 9.027 0 01-1.306-1.652m0 0l4.138-3.448M4.33 16.712a9.014 9.014 0 010-9.424m4.138 5.976a3.765 3.765 0 010-2.528m0 0c.181-.506.475-.982.88-1.388a3.736 3.736 0 011.388-.88m-2.268 2.268L4.33 7.288m6.406 1.18L7.288 4.33m0 0a9.024 9.024 0 00-1.652 1.306A9.025 9.025 0 004.33 7.288"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p />
        </div>
      </div>
    </div>
  </div>
</div>
`;
